fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,497 @@
+            embeddings, sample_ids, label_ids = self.compute_embeddings(
+                self._dataset
+            )
+            self.add_to_index(embeddings, sample_ids, label_ids)
+        self.find_duplicates(self._leak_threshold)
+        duplicates = self.duplicates_view()
+
+        to_remove = []
+        for sample_id, neighbors in self.neighbors_map.items():
+            remove_sample = True
+            sample_split = self._id2split(sample_id, self.split_views)
+            for n in neighbors:
+                if not (
+                    self._id2split(n[0], self.split_views) == sample_split
+                ):
+                    remove_sample = False
+                    if sample_split == 0:
+                        neighbor_sample = self._dataset.select([n[0]]).first()
+                        neighbor_sample.tags.append(f"leak with {sample_id}")
+                        neighbor_sample.save()
+
+            if remove_sample:
+                to_remove.append(sample_id)
+
+        duplicates = duplicates.exclude(to_remove)
+
+        return duplicates
+
+
+###
+
+### HASH BACKEND
+
+_HASH_METHODS = ["filepath", "image"]
+
+
+class LeakySplitsHashConfig(fob.BrainMethodConfig, LeakySplitsConfigInterface):
+    """
+    Args:
+        hash_field (None): string, field to write hashes into
+    """
+
+    def __init__(
+        self,
+        split_views=None,
+        split_field=None,
+        split_tags=None,
+        method="filepath",
+        hash_field=None,
+        **kwargs,
+    ):
+        self._method = method
+        self.hash_field = hash_field
+        LeakySplitsConfigInterface.__init__(
+            self,
+            split_views=split_views,
+            split_field=split_field,
+            split_tags=split_tags,
+        )
+        fob.BrainMethodConfig.__init__(self, **kwargs)
+
+    @property
+    def method(self):
+        return self._method
+
+
+class LeakySplitsHash(fob.BrainMethod):
+    def initialize(self, samples, brain_key):
+        return LeakySplitsHashIndex(
+            samples, self.config, brain_key, backend=self
+        )
+
+
+class LeakySplitsHashIndex(fob.BrainResults, LeakySplitIndexInterface):
+    """ """
+
+    def __init__(self, samples, config, brain_key, backend):
+        fob.BrainResults.__init__(
+            self, samples, config, brain_key, backend=backend
+        )
+        LeakySplitIndexInterface.__init__(self)
+        self._hash2ids = defaultdict(list)
+        self.split_views = _to_views(
+            samples,
+            self.config.split_views,
+            self.config.split_field,
+            self.config.split_tags,
+        )
+        self._dataset = samples._dataset
+        self._compute_hashes(samples)
+
+    @property
+    def _hash_function(self):
+        if self.config.method == "filepath":
+            return fou.compute_filehash
+
+        elif self.config.method == "image":
+            return LeakySplitsHashIndex._image_hash
```

Comment:
I could be missing it, but I don't see how `hash_size` propagates into `LeakySplitsHashIndex` to override the default

Reply:
It doesn't, it's internal. I think it may be best to remove it entirely, but I can add control for it. What do you think?

Reply:
same as above